### ISSUE TITLE
AsyncGenerator has no Symbol.asyncIterator

### ISSUE THREAD

**Issue by anywhichway** [babel/babel#7467]
Choose one: is this a bug report or feature request? bug

### Input Code

This is your generated code, not mine. Seems like generator is improperly created.

```js
function _asyncIterator(iterable) { // debugger shows: iterable = AsyncGenerator {_invoke: ƒ}
  if (typeof Symbol === "function") {
    if (Symbol.asyncIterator) { // this test succeeds
      var method = iterable[Symbol.asyncIterator]; // method ends up being undefined
      if (method != null) return method.call(iterable);
    } // so falls through
    if (Symbol.iterator) { // so this passes
      return iterable[Symbol.iterator](); // an attempt to call this is made, but it is also undefined, throws
    }
  }
  throw new TypeError("Object is not async iterable");
}
```

### Babel/Babylon Configuration (.babelrc, package.json, cli command)

Transpiled using your website Try It Out: Presets: ES2016, stage-0, stage-1, stage-2, stage-3

### Expected Behavior

iterable[Symbol.asyncIterator] should not be undefined

### Current Behavior

iterable[Symbol.asyncIterator] is undefined

### Context

The code runs un-transpiled in current versions of Chrome, Edge, Firefox. The only reason we are transpiling is to get the code to run in Node.js, which does not yet support async generators.

### Your Environment

Presets: ES2016, stage-0, stage-1, stage-2, stage-3
| software         | version(s)
| ---------------- | -------
| Babel            |  "@babel/polyfill": "^7.0.0-beta.40", code transpiled using your website Try It Out
| Babylon          | 
| node             |  9.7.1
| npm              | 
| Operating System |  Windows 10 Pro, 64bit,

**Comment by anywhichway** [comment]
Thanks for so quickly taking the time to build a test case. We just found the bug in our code after reviewing your test. Although it operates properly in Chrome, Firefox and Edge, our code below had a flaw, it needs an `await` before each recursion to work properly with transpiled code!

```javascript
async function* pipe(functions,arg=null,recursing) {
	if(!pipe.recursing) pipe.END = false;
	let next = functions[0];
	if(!next || pipe.END) return;
	if(Object.getPrototypeOf(next).constructor===Object.getPrototypeOf(function(){}).constructor) {
		const value = next(arg);
		if(value!==undefined) {
			if(functions.length===1) yield value;
			else yield* pipe(functions.splice(0,1),value,true);
		}
	} else if(Object.getPrototypeOf(next).constructor===Object.getPrototypeOf(async function(){}).constructor) {
		const value = await next(arg);
		if(value!==undefined) {
			if(functions.length===1) yield value;
			else yield* pipe(functions.splice(0,1),value,true);
		}
	} else if(Object.getPrototypeOf(next).constructor===Object.getPrototypeOf(function*(){}).constructor) {
		functions.splice(0,1);
		for (let value of next(arg)) {
			if(value!==undefined) {
				if(functions.length===0) yield value; 
				else yield* pipe(functions,value,true);
			}
			if(pipe.END) return;
		}
	} else if(Object.getPrototypeOf(next).constructor===Object.getPrototypeOf(async function*(){}).constructor) {
		functions.splice(0,1);
		for await (let value of next(arg)) {
			if(value!==undefined) {
				if(functions.length===0) yield value;
				yield* pipe(functions,value,true);
			}
			if(pipe.END) return;
		}
	} else {
		throw new TypeError("Unsupported function type passed to pipe")
	}
	return;
}
```

**Comment by nicolo-ribaudo** [comment]
Do you mean that you need to use `yield* await pipe()` instead of `yield* pipe()`? If it works in every browser, at 99% it is a bug with Babel.

**Comment by anywhichway** [comment]
@nicolo-ribaudo yes, "yield* await pipe()" ... and it only fixed part of the issue. the entire code base it at https://github.com/anywhichway/anywhichway ... having a hard time creating a small sample. I discovered the --harmony flag would work with Node, so I don't even have to transpile any more.  Should I re-open?

**Comment by maparent** [comment]
FYI: I have had the same problem using `for await (... of someAsyncGenerator)`
I'm using preset-env targets: node: current
And it works fine in node 9 or 10, where the Symbol.asyncIterator is defined on the generator,
but fails in node 8, where it is not defined.
This is using @babel 7.0.0
I hope that's helpful.

**Comment by Rockettomatoo** [comment]
Any updates on this?

I have an even simpler example on how to reproduce this error:
```js
async function* counter(max) {
  let count = 0;
  while(count < max) {
    yield await Promise.resolve(count++)
  }
}

(async () => {
  for await(const num of counter(3)) {
    console.log(num)
  }
})();
```
This throws `TypeError: Object is not iteratable`.
If I check `counter(3)[Symbol.asyncIterator]`, its `undefined`.


Out of legacy reasons (and [meteor.js](https://meteor.com)) we are still using node `8.15.1`.

@babel/runtime `v.7.4.5`
babel-plugin-transform-decorators-legacy `v.1.3.5`
```
// .babelrc
{
  "presets": ["env"],
  "plugins": ["decorators-legacy"]
}
```

**Comment by octet-stream** [comment]
I had similar problems on Node 8.
Switching from `@babel/runtime` to `@babel/runtime-corejs3` [fixed](https://github.com/octet-stream/form-data/blob/202b39a41995fe2e2df6e07a98d53b67be0f5c94/.babelrc.js#L3-L5) it for me.
